File: Daily/PY/Leetcode1720-解码异或后的数组.py
from typing import List
class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        n = len(encoded)
        res = [0] * (n + 1)
        res[0] = first
        for i in range(1,n + 1):
            res[i] = encoded[i - 1] ^ res[i - 1]
        return res

File: Leetcode/Rust/Leetcode1720-解码异或后的数组.rs
impl Solution {
    pub fn decode(encoded: Vec<i32>, first: i32) -> Vec<i32> {
        let n = encoded.len()  + 1;
        let mut res = vec![0;n];
        res[0] = first;
        for i in 1..n{
            res[i] = res[i - 1] ^ encoded[i - 1];
        }
        res
    }
}